templates/taskcat-ci-entrypoint.template.yaml

```diff
@@ -0,0 +1,519 @@
+                Resource: !GetAtt ArtifactBucket.Arn
+              - Sid: GetArtifactBucketObjects
+                Effect: Allow
+                Action: s3:GetObject
+                Resource: !Sub ${ArtifactBucket.Arn}/*
+              - Sid: GetSecrets
+                Effect: Allow
+                Action: secretsmanager:GetSecretValue
+                Resource: !Ref Secrets
+      ManagedPolicyArns:
+        - !Sub arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+  GitMergeFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Code:
+        S3Bucket: !If [UsingDefaultBucket, !Sub '${QSS3BucketName}-${AWS::Region}', !Ref QSS3BucketName]
+        S3Key: !Sub ${QSS3KeyPrefix}functions/packages/git-merge/lambda.zip
+      Description: Merge GitHub branches
+      Handler: index.handler
+      Role: !GetAtt GitMergeRole.Arn
+      Runtime: python3.9
+      Timeout: 30
+  CodeBuildServiceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action: sts:AssumeRole
+            Principal:
+              Service: codebuild.amazonaws.com
+      ManagedPolicyArns:
+        - !Sub arn:${AWS::Partition}:iam::aws:policy/AdministratorAccess
+  CodeBuild:
+    Type: AWS::CodeBuild::Project
+    Properties:
+      Description: !Sub >-
+        Submit build jobs for ${GitHubRepoName} as part of a CI/CD pipeline.
+      ServiceRole: !GetAtt CodeBuildServiceRole.Arn
+      Artifacts:
+        Type: CODEPIPELINE
+      Environment:
+        Type: LINUX_CONTAINER
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: aws/codebuild/standard:5.0
+        EnvironmentVariables:
+          - Name: ARTIFACT_BUCKET
+            Value: !Ref ArtifactBucket
+          - Name: PROJECT_NAME
+            Value: !Ref GitHubRepoName
+          - Name: PROJECT_OWNER
+            Value: !Ref GitHubOwner
+          - Name: SOURCE_BRANCH
+            Value: !Ref SourceBranch
+        PrivilegedMode: true
+      Visibility: PRIVATE
+      Source:
+        Type: CODEPIPELINE
+        GitCloneDepth: 1
+        BuildSpec: !Sub |
+          version: 0.2
+
+          env:
+            shell: bash
+            secrets-manager:
+              GITHUB_TOKEN: ${Secrets}:Token
+              GITHUB_USERNAME: ${Secrets}:Username
+              TASKCAT_OVERRIDES: arn:${AWS::Partition}:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:${GitHubRepoName}
+
+          phases:
+            install:
+              runtime-versions:
+                python: 3.x
+              commands:
+                - echo "Entered the install phase..."
+                - apt-get -qy update
+                - echo "Installing system dependencies..."
+                - apt-get -qy install zip gzip tar
+                - echo "Installing python dependencies..."
+                - pip3 install cfn-lint pre-commit taskcat yq
```

Comment:
Do you always want to install latest taskcat or tie it to a specific version?
Latest sometimes have breaking changes.

Reply:
Sounds good. I don't have strong opinions about any one approach over the other. Just wanted to make sure we have thought about it. I guess, we will learn what's better and always change if one approach is causing more trouble.